File: targets/jsts-web/src/vulns/vuln39_log.ts
import { Router } from "express";
export const router = Router();
router.get("/vuln39/note", (req, res) => {
  const text = String(req.query.text || "");
  // VULN:VULN-39:CWE-117:taint user input logged without neutralising CR/LF
  console.log("[audit] user note: " + text);
  res.json({ logged: true });
});

File: targets/python-web/vulns/vuln39_log.py
import logging
from flask import Blueprint, request, jsonify
bp = Blueprint("vuln39", __name__)
_log = logging.getLogger("audit")

@bp.get("/vuln39/note")
def note():
    text = request.args.get("text", "")
    # VULN:VULN-39:CWE-117:taint user input logged without neutralising CR/LF
    _log.warning("user note: " + text)
    return jsonify({"logged": True})
